websites/G/Grailed/presence.ts

```diff
@@ -0,0 +1,209 @@
+const presence = new Presence({
+    clientId: "786739998011293717"
+});
+
+let item: Element,
+item2: Element;
+
+const browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+        largeImageKey: "grailedlogo"
+    };
+
+    presenceData.startTimestamp = browsingStamp;
+
+    if (document.location.hostname == "www.grailed.com") {
+
+        if (document.location.pathname.includes("/listings/")) {
+            item = document.querySelector("body > div.container > div > div.show-listing > div.listing-details-wrapper > div.-listing-details-and-likes-wrapper > div.-listing-designer-title-size > h1");
+            
+            presenceData.details = "Viewing a listing:";
+            if (item.innerHTML.length > 128) {
+                presenceData.state = item.innerHTML.substring(0, 125) + "...";
+            } else {
+                presenceData.state = item.innerHTML;
+            } 
+            delete presenceData.smallImageKey;
+            presence.setActivity(presenceData);
+        
+        } else if (document.location.pathname.includes("mygrails")) {
+            
+            presenceData.details = "Browsing:";
+            presenceData.state = "My Grails";
+            
+            delete presenceData.smallImageKey;
+            presence.setActivity(presenceData);
+
+        } else if (document.location.pathname.includes("foryou")) {
+            
+            presenceData.details = "Viewing:";
+            presenceData.state = "For You";
+            
+            delete presenceData.smallImageKey;
+            presence.setActivity(presenceData);
+
+        } else if (document.location.href.includes("www.grailed.com/collections")) {
+            
+            if (document.location.href.includes("www.grailed.com/collections/")) {
```

Comment:
huh? Just check if it has /collectionsin href or pathname etc

Reply:
are you saying that i should get rid of the www.grailed part? because if so, there may be problems with users naming themselves in a certain way since grailed does not have /user/... in the url. so i believe this way would be more concise, but if you say so, i can change it. also, sorry, im pretty new to coding so any help would be appreciated :)

Reply:
are you saying about collections or about user profiles

Reply:
oh sorry im talking about the users. right after the .com is the user for example (https://www.grailed.com/grailedofficial). this also means that grailed.com/shop and grailed.com/shops are 2 completely different things. i guess i could do the check if the element exists, but honestly, i dont know much about that.

Reply:
Like I said above you can check if this element on ss exists 
![image](https://user-images.githubusercontent.com/9348108/102268848-1cd09d00-3f1c-11eb-8767-b62a8ef8acea.png)

Reply:
![code2](https://user-images.githubusercontent.com/75966600/102269887-20d7cd00-3ed2-11eb-9cb0-2717e48cbb99.JPG)

would something like this be what you're looking for?

Reply:
```const userName: HTMLDocument```
and yeah

Reply:
dumb question but why and how would i use HTMLDocument?

Reply:
HTMLElement*

Reply:
oh alright
![code3](https://user-images.githubusercontent.com/75966600/102272564-cd677e00-3ed5-11eb-8a79-daa6efaff769.JPG)
not sure, but is this fine?